pandas/core/algorithms.py

```diff
@@ -349,9 +354,10 @@ def factorize(values, sort=False, order=None, na_sentinel=-1, size_hint=None):
 
     table = hash_klass(size_hint or len(vals))
     uniques = vec_klass()
-    check_nulls = not is_integer_dtype(values)
-    labels = table.get_labels(vals, uniques, 0, na_sentinel, check_nulls)
+    check_nulls = (not is_integer_dtype(values)) and dropna
 
+    # the following line fills uniques:
```

Comment:
remove the comment (I generally like comments, but a more general comment or nothing is warranted here)